crates/biome_js_analyze/src/lint/performance/no_accumulating_spread.rs

```diff
@@ -56,26 +67,41 @@ declare_lint_rule! {
 }
 
 impl Rule for NoAccumulatingSpread {
-    type Query = Semantic<JsSpread>;
-    type State = ();
+    type Query = Semantic<SpreadOrAssign>;
+    type State = (TextRange, bool);
```

Comment:
Lets go ahead and make the `State` its own `struct` type instead of a tuple for readability.

eg:
```rust
struct FoundSpread {
    range: TextRange,
    is_spread: bool,
}
```
...
```rust
type State = FoundSpread;
```